### PR TITLE
Rewrite iam_policy using boto3

### LINKED ISSUE #64394: Region us-gov-east-1 does not seem to be available for aws module boto.iam
##### SUMMARY

attempting to use iam_policy module in govcloud and it results in the error:

"Region us-gov-east-1 does not seem to be available for aws module boto.iam. If the region definitely exists, you may need to upgrade boto or extend with endpoints_path"

the same playbook works fine in us-gov-west

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

iam_policy

##### ANSIBLE VERSION

```
{
    "exception": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1572882827.2-191594499800419/AnsiballZ_iam_policy.py\", line 114, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1572882827.2-191594499800419/AnsiballZ_iam_policy.py\", line 106, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1572882827.2-191594499800419/AnsiballZ_iam_policy.py\", line 49, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/tmp/ansible_iam_policy_payload_1NDwn0/__main__.py\", line 351, in <module>\n  File \"/tmp/ansible_iam_policy_payload_1NDwn0/__main__.py\", line 325, in main\n  File \"/tmp/ansible_iam_policy_payload_1NDwn0/ansible_iam_policy_payload.zip/ansible/module_utils/ec2.py\", line 340, in connect_to_aws\nansible.module_utils.ec2.AnsibleAWSError: Region us-gov-east-1 does not seem to be available for aws module boto.iam. If the region definitely exists, you may need to upgrade boto or extend with endpoints_path\n",
    "_ansible_no_log": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1572882827.2-191594499800419/AnsiballZ_iam_policy.py\", line 114, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1572882827.2-191594499800419/AnsiballZ_iam_policy.py\", line 106, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1572882827.2-191594499800419/AnsiballZ_iam_policy.py\", line 49, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/tmp/ansible_iam_policy_payload_1NDwn0/__main__.py\", line 351, in <module>\n  File \"/tmp/ansible_iam_policy_payload_1NDwn0/__main__.py\", line 325, in main\n  File \"/tmp/ansible_iam_policy_payload_1NDwn0/ansible_iam_policy_payload.zip/ansible/module_utils/ec2.py\", line 340, in connect_to_aws\nansible.module_utils.ec2.AnsibleAWSError: Region us-gov-east-1 does not seem to be available for aws module boto.iam. If the region definitely exists, you may need to upgrade boto or extend with endpoints_path\n",
    "changed": false,
    "module_stdout": "",
    "rc": 1,
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"
}

```

##### CONFIGURATION

```no output
```

##### OS / ENVIRONMENT

use iam_policy with the region targeting us-gov-east-1, in our case we just wanted to create a policy.

```

```

```
- name: Create IAM Policy for the Audit S3 bucket
  iam_policy:
    iam_type: role
    iam_name: "some_name"
    region: "{{ a var that results in us-gov-east-1 }}"
    policy_name: some_policy_name
    policy_json: "{{ lookup('stuff for policy json') }}"
    state: present

```

##### EXPECTED RESULTS

i expected the same results as with gov west, which was successful

##### ACTUAL RESULTS

the error described above

```
have pasted above
```

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Reworked iam_policy
- uses AnsibleAWSModule and boto3
- adds support for check_mode
- deprecates skip_duplicates defaulting to true
- deprecates policy_document option (recommends using policy_json + lookup() )

Fixes #64394

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/amazon/iam_policy.py

##### ADDITIONAL INFORMATION

Attempt to revive #50731 it would be a shame to see @sbj-ss's hard work go to waste.

### CODE REVIEW

File: docs/docsite/rst/porting_guides/porting_guide_2.10.rst [hunk 1]
```diff
@@ -63,10 +63,14 @@ The following functionality will be removed in Ansible 2.14. Please update updat
 * :ref:`ec2_key <ec2_key_module>`: the ``wait`` option will be removed. It has had no effect since Ansible 2.5.
 * :ref:`ec2_key <ec2_key_module>`: the ``wait_timeout`` option will be removed. It has had no effect since Ansible 2.5.
 * :ref:`ec2_lc <ec2_lc_module>`: the ``associate_public_ip_address`` option will be removed. It has always been ignored by the module.
+* :ref:`iam_policy <iam_policy_module>`: the ``policy_document`` option will be removed. To maintain the existing behaviour use the ``policy_json`` option read the file using the ``lookup`` plugin
```
Comment (samccann) [inline docs/docsite/rst/porting_guides/porting_guide_2.10.rst]:
```suggestion
* :ref:`iam_policy <iam_policy_module>`: the ``policy_document`` option will be removed. To maintain the existing behaviour use the ``policy_json`` option to read the file using the ``lookup`` plugin.
```

File: docs/docsite/rst/porting_guides/porting_guide_2.10.rst [hunk 2]
```diff
@@ -63,10 +63,14 @@ The following functionality will be removed in Ansible 2.14. Please update updat
...
+
 
 The following functionality will change in Ansible 2.14. Please update update your playbooks accordingly.
 
 * The :ref:`docker_container <docker_container_module>` module has a new option, ``container_default_behavior``, whose default value will change from ``compatibility`` to ``no_defaults``. Set to an explicit value to avoid deprecation warnings.
+* :ref:`iam_policy <iam_policy_module>`: the default value for the ``skip_duplicates`` option will change from ``true`` to ``false``.  To maintain the existing behaviour explicitly set it to ``true``
```
Comment (samccann) [inline docs/docsite/rst/porting_guides/porting_guide_2.10.rst]:
```suggestion
* :ref:`iam_policy <iam_policy_module>`: the default value for the ``skip_duplicates`` option will change from ``true`` to ``false``.  To maintain the existing behavior explicitly set it to ``true``.
```

File: lib/ansible/modules/cloud/amazon/iam_policy.py [hunk 3]
```diff
@@ -40,6 +39,8 @@
     description:
       - The path to the properly json formatted policy file.
       - Mutually exclusive with I(policy_json).
+      - This option has been deprecated and will be removed in 2.14.  The existing behaviour can be
```
Comment (samccann) [inline lib/ansible/modules/cloud/amazon/iam_policy.py]:
```suggestion
      - This option has been deprecated and will be removed in 2.14.  The existing behavior can be
```

File: docs/docsite/rst/porting_guides/porting_guide_2.10.rst (hunk 1 above)
Comment (tremble) [reply docs/docsite/rst/porting_guides/porting_guide_2.10.rst]: @samccann How do you feel about this as an alternative:
```
To maintain the existing behavior use the ``policy_json`` option and read the file with the ``lookup`` plugin.
```

I think it's a little clearer?

Comment (samccann) [reply docs/docsite/rst/porting_guides/porting_guide_2.10.rst]: sounds good.

**Review by samccann — APPROVED** [review]
docs portion lgtm

**Review by jillr — APPROVED** [review]
LGTM, thanks very much @tremble!

**Comment by sbj-ss** [comment]
Thanks for keeping this alive. Sorry to say this but I'm now very far from ansible after changing my job this spring. It's great to see that my changes made it to the upstream :)

**Comment by tremble** [comment]
Thanks for the work you put in originally @sbj-ss !

**Pull request merged**